app/models/case/sar/offender.rb

```diff
@@ -104,4 +112,24 @@ def current_team_and_user_resolver
   def is_offender_sar?
     true
   end
+
+  # This method is here to fix an issue with the gov_uk_date_fields
+  # where the validation fails since the internal list of instance
+  # variables lacks the date_of_birth field from the json properties
+  #     NoMethodError: undefined method `valid?' for nil:NilClass
+  #     ./app/state_machines/configurable_state_machine/machine.rb:256
+  #
+  # Only reassign dates that have not changed.
+  def reassign_gov_uk_dates
+    reassign = GOV_UK_DATE_FIELDS.map(&:to_s) - self.changed
+    reassign.each do |field|
+      self.send("#{field}=", self.read_attribute(field))
+    end
+  end
+
+  # User can add data requests at any point in the workflow, but
+  # the case state should not change
+  def allow_waiting_for_data_state?
+    self.current_state == 'data_to_be_requested'
+  end
 end
```

Comment:
@Polsonby This is the workaround I mentioned regarding preventing GovUKDateFields from throwing the 'no method error valid? on nil class'. I believe it occurs when creating an object using the ActiveRecord relationship. In my case doing `offender_sar.data_requests.create!(...)` (IN A TEST!!) resulted in the `valid? on nil:class` error. This is my proposed work around. I have lost too many hours to this issue.

Reply:
Same here! Looks cool - nice job, I'm sure it will work better than my previous "fix"